src/test/java/com/google/sps/IsValidGameNameTest.java

```diff
@@ -0,0 +1,189 @@
+
+    Entity userEntity2;
+    ArrayList<String> gameNames2;
+
+    List<Entity> userList = new ArrayList<Entity>();
+    userEntity2 = new Entity("user");
+    gameNames2 = new ArrayList<>();
+
+    userEntity2.setProperty("gameNames",gameNames2);
+    userEntity2.setProperty("userID",userID2);
+
+    userList.add(userEntity2);
+
+    boolean actual = query.IsValidGameName("game1", userEntity2);
+    boolean expected = true;
+
+    Assert.assertEquals(expected, actual);
+  }
+
+  // given a game name the user already has, the function should return false 
+  @Test
+  public void duplicateGameName() {
+
+    Entity userEntity1;
+    ArrayList<String> gameNames1;
+
+    List<Entity> userList = new ArrayList<Entity>();
+    userEntity1 = new Entity("user");
+    gameNames1 = new ArrayList<>();
+
+    
+    gameNames1.add("game1");
+    gameNames1.add("game2");
+    gameNames1.add("Game3");
+    userEntity1.setProperty("gameNames",gameNames1);
+    userEntity1.setProperty("userID",userID1);
+
+    userList.add(userEntity1);
+
+    boolean actual = query.IsValidGameName("game1", userEntity1);
+    boolean expected = false;
+
+    Assert.assertEquals(expected, actual);
+  }
+
+  // when the user has games, but the given game name is new, the function should return true
+  @Test
+  public void newGameName() {
+
+    Entity userEntity1;
+    ArrayList<String> gameNames1;
+
+    List<Entity> userList = new ArrayList<Entity>();
+    userEntity1 = new Entity("user");
+    gameNames1 = new ArrayList<>();
+
+    
+    gameNames1.add("game1");
+    gameNames1.add("game2");
+    gameNames1.add("Game3");
+    userEntity1.setProperty("gameNames",gameNames1);
+    userEntity1.setProperty("userID",userID1);
+
+    userList.add(userEntity1);
+
+    boolean actual = query.IsValidGameName("game4", userEntity1);
+    boolean expected = true;
+
+    Assert.assertEquals(expected, actual);
+  }
+
+  // when the user has games, and the given name is a duplicate name with different capitalization, the function should return true
+  @Test
+  public void sameNameDifferentCapitalization() {
+
+    Entity userEntity1;
+    ArrayList<String> gameNames1;
+
+    List<Entity> userList = new ArrayList<Entity>();
+    userEntity1 = new Entity("user");
+    gameNames1 = new ArrayList<>();
+
+    
+    gameNames1.add("game1");
+    gameNames1.add("game2");
+    gameNames1.add("Game3");
+    userEntity1.setProperty("gameNames",gameNames1);
+    userEntity1.setProperty("userID",userID1);
+
+    userList.add(userEntity1);
+
+    boolean actual = query.IsValidGameName("Game1", userEntity1);
+    boolean expected = true;
+
+    Assert.assertEquals(expected, actual);
+  }
+
+  // given a blank game name, function should return false 
+  @Test
+  public void emptyGameName() {
+
+    Entity userEntity1;
+    ArrayList<String> gameNames1;
+
+    List<Entity> userList = new ArrayList<Entity>();
+    userEntity1 = new Entity("user");
+    gameNames1 = new ArrayList<>();
+
```

Comment:
extra space here

Reply:
same thing of there being an extra space in the other test functions

Reply:
that should be removed